src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -8287,7 +8138,124 @@ VOID MethodTableBuilder::PlaceThreadStaticFields()
 //
 // Place instance fields
 //
-VOID    MethodTableBuilder::PlaceInstanceFields(MethodTable ** pByValueClassCache)
+VOID MethodTableBuilder::PlaceInstanceFields(MethodTable** pByValueClassCache)
+{
+    MethodTable* pParentMT = GetParentMethodTable();
+    bool hasNonTrivialParent = pParentMT && !pParentMT->IsObjectClass() && !pParentMT->IsValueTypeClass();
+
+    EEClassLayoutInfo::NestedFieldFlags nestedFieldFlags =
+        EEClassLayoutInfo::GetNestedFieldFlags(
```

Comment:
We don't need to call this before `if (bmtLayout->LayoutType == EEClassLayoutInfo::LayoutType::Auto)` line unless the current target might need to have the isAlign8 bit set, and doing so adds an extra loop through all of the fields of the type. I'd like to see
1. The NestedFieldFlags::Align8 enum entry only exist if the FEATURE_64_BIT_ALIGNMENT or FEATURE_DOUBLE_ALIGNMENT_HINT defines are set.
2. If we don't support Align8, we shouldn't be calling GetNestedFieldFlags if the LayoutKind is specified as Auto.